### PR TITLE
Recursive Extend Model Default Options for getterMethods and setterMethods

### PR DESCRIPTION
**Pull request by maennchen**
If I define default getters and setters and create a Modle with some getter / setters themselves, the default ones won't be added.

I think we should handle getters & setters the same way we handle classMethods & instanceMethods.

### CODE REVIEW

**Comment by janmeier** [comment]
Good idea, it should definitely be merged. Needs a test to avoid regressions :-). 

I'm wondering whether this block of code could just be replaced with `_.merge`

**Comment by maennchen** [comment]
I'm not sure where to do the test and what exactly to test. I'm not familiar with this library. I started to use it a few minutes ago. ;-)

**Comment by janmeier** [comment]
Probably somewhere around here https://github.com/sequelize/sequelize/blob/master/test/integration/sequelize.test.js#L454. Try to set both global and local options and ensure that both functions are available 

Welcome btw :)

**Comment by janmeier** [comment]
ping @maennchen

**Comment by maennchen** [comment]
How can I test the setters?
https://github.com/maennchen/sequelize/blob/743123868290839c9dacde770f1e9abaed79028f/test/integration/sequelize.test.js#L491

**Comment by maennchen** [comment]
@janmeier All the tests are failing. I don't think I caused it. Especially because my changes should fail either all of the tests or none.
Can you have a look?

**Comment by mickhansen** [comment]
@maennchen easiest would probably just to use spies for all the getters and setters and verified that they are each called the appropriate number of times (once should suffice).

**Comment by maennchen** [comment]
The Getters are not a Problem, I can return things back there. I'll have a look at spies for the Getters.

**Comment by mickhansen** [comment]
@maennchen setterMethods/getterMethods get converted in to dynamic properties with get/set implemented so using spies for both would be the more robust test.

**Comment by maennchen** [comment]
@mickhansen I did everything with spies now.

File: test/integration/sequelize.test.js [hunk 1]
```diff
@@ -411,6 +411,115 @@ describe(Support.getTestDialectTeaser('Sequelize'), function() {
...
+      testEntity.customClassMethod();
+      testEntity.overrideClassMethod();
+
+      expect(typeof testEntity.defaultClassMethod).to.equal('function');
+      expect(typeof testEntity.customClassMethod).to.equal('function');
+      expect(typeof testEntity.overrideClassMethod).to.equal('function');
+
+      expect(defaultClassMethod).to.have.been.calledOnce;
+      expect(customClassMethod).to.have.been.calledOnce;
+      expect(overrideClassMethod.callCount).to.be.eql(0);
```
Comment (mickhansen) [inline test/integration/sequelize.test.js]: Can be express as `.not.to.have.been.called` i think.

Comment (maennchen) [inline test/integration/sequelize.test.js]: I tried. There was a strange error. It said something like "spy has to be a spy". (I'm on my mobile phone right now..)

Comment (mickhansen) [inline test/integration/sequelize.test.js]: Hm alright, not overly important.

**Comment by maennchen** [comment]
I don't get why the travis tests fail. Has this something to do with my changes?

**Comment by mickhansen** [comment]
@maennchen no i broke something on master ;p let me clean it up and you can rebase.

**Comment by mickhansen** [comment]
@maennchen master is green again if you could please rebase.

**Comment by maennchen** [comment]
@mickhansen MSSQL still fails -.-

**Comment by janmeier** [comment]
Thanks @maennchen

**Pull request merged**